cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -30,6 +30,11 @@ import (
 type MasterConfiguration struct {
 	metav1.TypeMeta
 
+	// Metadata holds metadata about the this configuration object.
```

Comment:
So I'm ok with adding meta-data, but I do think clusterName should remain depending on the naming of the the object. 

If we call it "ClusterConfiguration" than I would be ok with nixing cluster name, but right now it's master and there can be many so nixing the clustername doesn't make a ton of sense.  At least from a single PR perspective.

Reply:
This isn't using `metadata.Name` (as I'd first thought) but instead using `metadata.ClusterName`, which I think is supposed to be consistently filled in for all resources in the same cluster. I'm not sure any installation actually does that in practice though. Do we have any idea whether this is following precedent or setting it?